### ISSUE TITLE
error with `agat_sp_compare_two_annotations.pl`

### ISSUE THREAD

**Issue by jantusan** [NBISweden/AGAT#216]
I tried to compare to GTFs created with gffread or AGAT. When running `agat_sp_compare_two_annotations.pl` I got the following error `Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.`

<details>
  <summary>full command and stdout</summary>

 ```
$ agat_sp_compare_two_annotations.pl -gff1 ITAG4.1_gene_models_Mt_Pt.gtf -gff2 ITAG4.1_gene_models_Mt_Pt.gffread.gtf 
********************************************************************************
*                              - Start parsing -                               *
********************************************************************************
-------------------------- parse options and metadata --------------------------
=> Accessing the feature level json files
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level1.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level2.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level3.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_spread.json file
=> Attribute used to group features when no Parent/ID relationship exists:
	* locus_tag
	* gene_id
=> merge_loci option deactivated
=> Accessing Ontology
	No ontology accessible from the gff file header!
	We use the SOFA ontology distributed with AGAT:
		/home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo
	Read ontology /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo:
		4 root terms, and 2472 total terms, and 1436 leaf terms
	Filtering ontology:
		We found 1757 terms that are sequence_feature or is_a child of it.
-------------------------------- parse features --------------------------------
Parsing:   0% [*                                                     ]=> GFF parser version used: 2
Parsing: 100% [===================================================== ]D 0h03m26s
********************************************************************************
*                               - End parsing -                                *
*                             done in 209 seconds                              *
********************************************************************************

********************************************************************************
*                               - Start checks -                               *
********************************************************************************
---------------------------- Check1: feature types -----------------------------
----------------------------------- ontology -----------------------------------
All feature types in agreement with the Ontology.
------------------------------------- agat -------------------------------------
AGAT can deal with all the encountered feature types (3rd column)
------------------------------ done in 0 seconds -------------------------------

------------------------------ Check2: duplicates ------------------------------
None found
------------------------------ done in 0 seconds -------------------------------

-------------------------- Check3: sequential bucket ---------------------------
None found
------------------------------ done in 9 seconds -------------------------------

--------------------------- Check4: l2 linked to l3 ----------------------------
No problem found
------------------------------ done in 1 seconds -------------------------------

--------------------------- Check5: l1 linked to l2 ----------------------------
No problem found
------------------------------ done in 0 seconds -------------------------------

--------------------------- Check6: remove orphan l1 ---------------------------
We remove only those not supposed to be orphan
None found
------------------------------ done in 0 seconds -------------------------------

------------------------------ Check7: check cds -------------------------------
No problem found
------------------------------ done in 0 seconds -------------------------------

----------------------------- Check8: check exons ------------------------------
No exons created
No exons locations modified
No supernumerary exons removed
No level2 locations modified
------------------------------ done in 19 seconds ------------------------------

------------------------------ Check9: check utrs ------------------------------
No UTRs created
No UTRs locations modified
No supernumerary UTRs removed
------------------------------ done in 12 seconds ------------------------------

------------------------ Check10: all level2 locations -------------------------
No problem found
------------------------------ done in 17 seconds ------------------------------

------------------------ Check11: all level1 locations -------------------------
No problem found
------------------------------ done in 2 seconds -------------------------------

---------------------- Check12: remove identical isoforms ----------------------
None found
------------------------------ done in 0 seconds -------------------------------
********************************************************************************
*                                - End checks -                                *
*                              done in 60 seconds                              *
********************************************************************************

=> OmniscientI total time: 269 seconds
********************************************************************************
*                              - Start parsing -                               *
********************************************************************************
-------------------------- parse options and metadata --------------------------
=> Accessing the feature level json files
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level1.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level2.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_level3.json file
	Using standard /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/features_spread.json file
=> Attribute used to group features when no Parent/ID relationship exists:
	* locus_tag
	* gene_id
=> merge_loci option deactivated
=> Accessing Ontology
	No ontology accessible from the gff file header!
	We use the SOFA ontology distributed with AGAT:
		/home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo
	Read ontology /home/u1864855/miniconda3/lib/site_perl/5.26.2/auto/share/dist/AGAT/so.obo:
		4 root terms, and 2472 total terms, and 1436 leaf terms
	Filtering ontology:
		We found 1757 terms that are sequence_feature or is_a child of it.
-------------------------------- parse features --------------------------------
Parsing:   0% [*                                                     ]=> GFF parser version used: 2
Parsing: 100% [======================================================]D 0h02m29s
********************************************************************************
*                               - End parsing -                                *
*                             done in 155 seconds                              *
********************************************************************************

********************************************************************************
*                               - Start checks -                               *
********************************************************************************
---------------------------- Check1: feature types -----------------------------
----------------------------------- ontology -----------------------------------
All feature types in agreement with the Ontology.
------------------------------------- agat -------------------------------------
AGAT can deal with all the encountered feature types (3rd column)
------------------------------ done in 0 seconds -------------------------------

------------------------------ Check2: duplicates ------------------------------
None found
------------------------------ done in 0 seconds -------------------------------

-------------------------- Check3: sequential bucket ---------------------------
Nothing to check as sequential bucket!
------------------------------ done in 0 seconds -------------------------------

--------------------------- Check4: l2 linked to l3 ----------------------------
34899 cases fixed where L3 features have parent feature(s) missing
------------------------------ done in 21 seconds ------------------------------

--------------------------- Check5: l1 linked to l2 ----------------------------
No problem found
------------------------------ done in 1 seconds -------------------------------

--------------------------- Check6: remove orphan l1 ---------------------------
We remove only those not supposed to be orphan
None found
------------------------------ done in 0 seconds -------------------------------

------------------------------ Check7: check cds -------------------------------
No problem found
------------------------------ done in 0 seconds -------------------------------

----------------------------- Check8: check exons ------------------------------
124 exons created that were missing
No exons locations modified
No supernumerary exons removed
No level2 locations modified
------------------------------ done in 24 seconds ------------------------------

------------------------------ Check9: check utrs ------------------------------
44959 UTRs created that were missing
No UTRs locations modified
No supernumerary UTRs removed
------------------------------ done in 20 seconds ------------------------------

------------------------ Check10: all level2 locations -------------------------
No problem found
------------------------------ done in 14 seconds ------------------------------

------------------------ Check11: all level1 locations -------------------------
No problem found
------------------------------ done in 2 seconds -------------------------------

---------------------- Check12: remove identical isoforms ----------------------
None found
------------------------------ done in 0 seconds -------------------------------
********************************************************************************
*                                - End checks -                                *
*                              done in 82 seconds                              *
********************************************************************************

=> OmniscientI total time: 237 seconds
GFF3 files parsed
GFF3 files sorted
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
Use of uninitialized value in numeric lt (<) at /home/u1864855/miniconda3/bin/agat_sp_compare_two_annotations.pl line 136.
(and a lot more lines like those)
```
</details>

I cheked  the code for line 136 of `agat_sp_compare_two_annotations.pl` and looks like this:
```perl
[…]
130   else{
131     my $continue = 1;
132     $one_more_round_signalA = undef;
133     while ( $continue ){
134       #  location A  -------------------------
135       #  location A bis                                     -------------------------
136       if(  $current_flattened_locations->[$#$current_flattened_locations][1] < $location1->[1] ){ # End Location A < Start Location A bis
137         $continue = undef;
138 
139         if(! $one_more_round_signalA){
140             print "Let save the result 1: ( A $overlap_A => $overlap_B)\n" if ($verbose);
[…]
 ```

I installed AGAT with conda.

**Comment by Juke34** [comment]
Could you send me over example files providing this error?

**Comment by Juke34** [comment]
Without your file I cannot help you more. I hope you found your way, don't hesitate to reopen the issue if you are still in trouble.

**Comment by EinarBaldvin** [comment]
Hi @Juke34 ,

I encountered the same issue, but when I reversed the order of the gff input files it finished normally. I can send you the gff files if needed.

Cheers,
Einar

**Comment by Juke34** [comment]
Hi,
Which version of agat were you using? Yes the files would be useful thanks

**Comment by EinarBaldvin** [comment]
Hi,

the version I am using, installed via conda:
agat                      0.9.2           pl5321hdfd78af_1

**Comment by Juke34** [comment]
Humm it sounds I should completely rewrite this script I'm not sure to be able to fix it has it is.

**Comment by Juke34** [comment]
Fixed in AGAT v1.0.0
